### DISCUSSION TITLE
Minor changes to OG descriptions

### DISCUSSION THREAD

**Discussion by Mr-Bajs** [ihhub/fheroes2#6915] [General]
The descriptions of the buildings in the OG doesn't come with a uniform standard. Sometimes an increase is spelled numericaly and sometimes by words and sometimes a + or a - sign is added.

Coliseum, Rainbow and Tavern for example spells it out by words. 

OG Coliseum
![Coliseum](https://user-images.githubusercontent.com/93934125/228833193-3be1988d-7a37-446b-afa1-ae21d59a6224.png)

OG Rainbow
![Rainbow](https://user-images.githubusercontent.com/93934125/228833657-ffc2f35d-5765-417f-bd15-ff2c1b30440e.png)

OG Tavern
![Tavern](https://user-images.githubusercontent.com/93934125/228834028-2adc80b1-71b6-45b7-8059-d4809bd52878.png)

But other building spells it by numbers. Both Storm and Castle adds a + and a - sign but the well does not.

OG Storm
![Storm](https://user-images.githubusercontent.com/93934125/228834599-9679862e-9f0c-414e-9212-d8e8da10df51.png)

OG Castle
![Castle](https://user-images.githubusercontent.com/93934125/228834880-4ff4d083-d2f8-4685-8c74-3d74e0d305be.png)

OG Well
![Well](https://user-images.githubusercontent.com/93934125/228834946-7636b929-a2f8-457a-9901-ba151b621526.png)

It would be nice if it would be more uniform. Example comparison of the Coliseum.

![Coliseumexamples](https://user-images.githubusercontent.com/93934125/228844322-e834d3f4-069d-437e-af33-2fbd8bea3142.png)

I think v.2 looks the best even if the + and - signs are redundant in some way, but its used in the Storm and Castle in the OG. If it would be changed then it should be applied to artifacts as well, there having a + or - before the number makes it more clear if new artifacts where to be added that have negative or both positive and negative effects.

Mockup example:
![Example](https://user-images.githubusercontent.com/93934125/228845938-2c2117a3-8a5f-4a62-9af0-d40f389f5255.png)

Or just leave it as is.
Related PR: https://github.com/ihhub/fheroes2/pull/6909

**Comment by LeHerosInconnu** [comment]
Hello @Mr-Bajs,

I would rather go for version 2, with the sign and the number (example: "+2" and "-2"), and I think the sentences should be reworded so that it is not redundant.

For example:
"gives +2 attack"
Instead of:
"adds +2 attack"

**Reply by Mr-Bajs** [reply]
Hi @LeHerosInconnu 

The Storm is the only time i've seen the use of add in a description.

The Storms description should be changed in my mind into something like "_The Storm increases the castles defending spellcaster spellpower by +2_" or "_The Storm increases the spell power of the castles defending spellcaster by +2_"

**Reply by LeHerosInconnu** [reply]
@Mr-Bajs.

> Hi @LeHerosInconnu
> 
> The Storm is the only time i've seen the use of add in a description.
> 
> The Storms description should be changed in my mind into something like "_The Storm increases the castles defending spellcaster spellpower by +2_" or "_The Storm increases the spell power of the castles defending spellcaster by +2_"

I find "increases" and "by +2" still redundant.

How about something like this:
"The Storm gives +2 spell power to the defending spell caster."
"The Storm gives the defending caster +2 spell power."
"The Storm grants the defending caster +2 spell power."

**Reply by Branikolog** [reply]
Hi, @LeHerosInconnu !
"Increases" is used in many other descriptions, so I'm not strictly against of this word.
I'm OK with most variants containing numeric description of the bonus, but I would like to see the number and the attribute itself as close to each other, as it possible (within a single sentence).
And I personally like descriptions without "+" more, except castle towers descriptions (where "+5 bonus" looks quite relevant with the word "bonus"). I agree, that "+x something" is a little more informative, but stylistically I think text without "+" and "-" marks looks more pleasant to eye.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hi, @LeHerosInconnu !
> "Increases" is used in many other descriptions, so I'm not strictly against of this word. I'm OK with most variants containing numeric description of the bonus, but I would like to see the number and the attribute itself as close to each other, as it possible (within a single sentence). And I personally like descriptions without "+" more, except castle towers descriptions (where "+5 bonus" looks quite relevant with the word "bonus"). I agree, that "+x something" is a little more informative, but stylistically I think text without "+" and "-" marks looks more pleasant to eye.

It's not so much the word, but its use with the "+" sign.
"Increases" it's already "to add", so the "+" sign is redundant.
With "gives" it works better, because it can be "give more" or "give less".
"Grants" is even better, because for "gives", as a gift, it's always a give and not take action.

For players who don't read the texts too much, or who don't understand the language too much, "+2 spell power" is better (as you also wrote, "more informative", and I would also add "quickly").
All the information is grouped together in one place, preferably at the end of the sentence.
For the aesthetic side, I think it's okay, because the information given is clearer that way.

Also "+" and "-" signs with numbers are commonly used in the interface:
In fheroes2:

![Sign number 001](https://user-images.githubusercontent.com/43583994/229121594-c8ccd18a-d1f7-49fd-b228-7c868a0b344c.png)
![Sign number 002](https://user-images.githubusercontent.com/43583994/229121604-5ed8f02d-6049-412e-976c-688a8945e8b8.png)

**Reply by Branikolog** [reply]
@LeHerosInconnu 
In the example above I consider "+" as a short form of "add" in short calculation list. We also have not so much free space here. So it's OK to have "+" here.
I personally like the form without "+" mark within full literary description of an artifact or a town structure.

>It's not so much the word, but its use with the "+" sign.
>"Increases" it's already "to add", so the "+" sign is redundant.
>With "gives" it works better, because it can be "give more" or "give less".
>"Grants" is even better, because for "gives", as a gift, it's always a give and not take action.

I agree, that both "+" and "Increase" cannot be used together.
I personally find "increase" more pleasant to eye and it's also easier to propose inverse definition for "-something" effect. I mean pairs increase/decrease or increase/reduce looks more uniform semantically.

**Reply by Mr-Bajs** [reply]
> I agree, that both "+" and "Increase" cannot be used together. I personally find "increase" more pleasant to eye and it's also easier to propose inverse definition for "-something" effect. I mean pairs increase/decrease or increase/reduce looks more uniform semantically.

Sure without - and + looks better but with - and + is more informative at a quick glance.
And **increase** is used by default in most cases in OG and fheroes2, in grant is used instead then all instances of the use of increases should be changed to grant for the sake of looking uniform.




**You/The Hero**
Another thing i noticed, of this is of course a minor or no issue, and only noticed when doing translation but i'll bring it up anyway. The game sometimes refers to you the player and sometimes to the hero.

Seconday Skills refer as the skills belong to the hero, but Primary Skills refers to you (the player). In it context I assume everyone understands what the skills and that the Primary Skills belongs to the hero rather then You. But still it's a little weird.

Example Mysticism
![myst](https://user-images.githubusercontent.com/93934125/229289565-e58694af-f199-403f-84dd-5f478a708714.png)

Example Defense Skill
![defence](https://user-images.githubusercontent.com/93934125/229289865-e02e42a9-7dfd-4988-b3cd-28e796666c5d.png)

The Primary Skills (and its related descritopns such as Skills Points) descriptions  should might be changed into something like so it refers to that the skills belongs to the Hero and not the Player. 

Example suggestion:
_**The Hero's** defense skill is a bonus added to each creature's defense skill._ 
rather than
_**Your** defense skill is a bonus added to each creature's defense skill._

Course the when you have a hero selected you **are** the hero. So there isn't really an issue. But maybe rules should be implemented going forward when new assets are being added.

**Artifacts**
Arifacts desc and the events that happen when the hero picks up an artifact refers to you. That is might be inconsistent but i think that works fine though. The short stories that are tied when picking an artifact are better told through a seconday perspective, with You rather then The Hero.

**Reply by Branikolog** [reply]
Hi, @Mr-Bajs 

> Sure without - and + looks better but with - and + is more informative at a quick glance.

Both have pros and cons. That's why we're here discussing it. :) The result will be based on all opinions.

> Another thing i noticed, of this is of course a minor or no issue, and only noticed when doing translation but i'll bring it up anyway. The game sometimes refers to you the player and sometimes to the hero.

Oh. I find this suggestion nice. Indeed, all skills and attributes are referred to a particular hero (you, as a player, could have several heroes, so each of them has his own set of attributes).

>Seconday Skills refer as the skills belong to the hero, but Primary Skills refers to you (the player).

I think it would be nice to have these descriptions referred to a hero, not a player, to keep texts more consistent in this aspect.

**Reply by zenseii** [reply]
Hi all. Personally I see no problem with having variation, so sometimes "increase", "add" and "grant". This doesn't really have to be the same all over the game. The real issue imo is whether to write the unnecessary "+" or not. I think we can remove it.

**Comment by Mr-Bajs** [comment]
Sure variation isn't bad either.

I suggest a change into this

**Tavern**
_The Tavern increases morale for troops defending the castle by 1._

**The Coliseum**
_The Coliseum provides inspiring spectacles to defending troops, raising their morale by 2 during combat._

**The Rainbow**
 _The Rainbow increases the luck of the defending units by 2._

**The Storm**
 _The Storm increases the spell power of defending spell caster by 2._

Also the Statue should be changed from your income into the towns income. Makes its clear that the statues income is tied to the town and not the player.

**The Statue**
From this
 _The Statue increases your town's income by %{count} gold per day._
to this
 _The Statue increases the town's income by %{count} gold per day._

**Reply by Branikolog** [reply]
Everything seems good to me from the proposal above. 👍 
I'm only concerning about _the_ in rainbow description:

>The Rainbow
>The Rainbow increases the luck of _**the**_ defending units by 2.

Since all other descriptions don't have _the_ before _units_.

**Comment by Mr-Bajs** [comment]
Primary Skills into something like this, I suggest the second option, that makes its more clear that it isn't global effects.

**Attack Skill**
1. _The hero's attack skill is a bonus added to each creature's attack skill._
2. _The hero's attack skill is a bonus added to each of the hero's creatures' attack skill._

**Defense Skill**
1. _The hero's defense skill is a bonus added to each creature's defense skill._
2. _The hero's defense skill is a bonus added to each of the hero's creatures' defense skill._

**Spell Power**
1. _The hero's spell power determines the length or power of a spell._
2. _The hero's spell power determines the duration or power of the hero's spells._

**Knowledge**
1. _The hero's knowledge determines how many spell points your hero may have. Under normal circumstances, a hero is limited to 10 spell points per level of knowledge._
2. _The hero's knowledge determines how many spell points the hero may have. Under normal circumstances, a hero is limited to 10 spell points per level of knowledge._

**Reply by zenseii** [reply]
Sounds good to me. Just note that it should be `each of the hero's creatures'` and not `creature's`.

I would also suggest to change to duration rather than length for the spell power description, but this isn't critical.

**Reply by Mr-Bajs** [reply]
Okey. Sure, duration is better.

**Reply by Branikolog** [reply]
I personally prefer shorter variants for each description (1, 1, 1, 2).

**Reply by zenseii** [reply]
@Mr-Bajs 

> I would also suggest to change to duration rather than length for the spell power description, but this isn't critical.

And now I see that I'm arguing against myself 🤣

**Comment by Mr-Bajs** [comment]
Secondary skills are mostly fine. But might be changed from _your hero's_ into _the hero's_ but it's fine as it is. But if multiplayer is improved then you might wanna view your ally hero, then it make more sense if its 'the hero' and not 'your hero'. But change is probably unnecessary.

A few secondary skills should be changed to make it clear that the skill is tied to the hero and not the player.

**Diplomacy**
**OG:** Diplomacy allows you to negotiate with monsters who are weaker than your group.
**Suggestion:** Diplomacy allows the hero to negotiate with monsters that are weaker than the hero's army.

Eagle Eye refers to the hero as 'him' which is wierd if the hero is a female such as a witch. Should be neutral.

**Eagle Eye**
**OG:** Eagle Eye gives your hero a %{count} percent chance to learn any given 1st or 2nd level enemy spell used against him in a combat.
**Suggestion:** Eagle Eeye gives the hero a %{count} percent chance to learn any given 1st or 2nd level enemy spell used against the hero in a combat.

**Pathfinding**
**OG:** Pathfinding reduces the movement penalty for rough terrain by %{count} percent.
**Suggestion:** Pathfinding reduces the hero's movement penalty for rough terrain by %{count} percent.

**Archery**
**OG:** Archery ncreases the damage done by range attacking creatures by %{count} percent.
**Suggestion:** Archery increases the damage done by the hero's range attacking creatures by %{count} percent.

**Comment by LeHerosInconnu** [comment]
Hello everyone,

Okay, let's go with "verb" and "by X" for the description.
It will always be possible to add a quick info in brackets for example like in the original game (to be improved because not all artifacts have this quick info in the original game).

In original game:

![Quick info brackets](https://user-images.githubusercontent.com/43583994/229544960-62d60d17-1645-415c-9fad-4f09149c192d.png)

"You/The Hero"
Yes, it is better to refer to the hero.

All the changes concerning the buildings, secondary skills and primary skills seem to be appropriate.
